ember-model.js

```diff
@@ -1371,18 +1429,22 @@ Ember.hasMany = function(type, options) {
 
   var meta = { type: type, isRelationship: true, options: options, kind: 'hasMany', getType: getType};
 
-  return Ember.computed(function(propertyKey, newContentArray, existingArray) {
-    type = meta.getType(this);
-    Ember.assert("Type cannot be empty", !Ember.isEmpty(type));
+  return Ember.Model.computed({
+    get: function(propertyKey) {
+      type = meta.getType(this);
+      Ember.assert("Type cannot be empty", !Ember.isEmpty(type));
 
-    var key = options.key || propertyKey;
-
-    if (arguments.length > 1) {
-      return existingArray.setObjects(newContentArray);
-    } else {
+      var key = options.key || propertyKey;
       return this.getHasMany(key, type, meta, this.container);
+    },
+    set: function(propertyKey, newContentArray, existingArray) {
+      //SWRVE-CHANGES
+      if (existingArray !== newContentArray) {
```

Comment:
I've updated that to use isEquals

On Thu, Oct 1, 2015 at 2:41 PM, Robert Clancy <notifications@github.com>
wrote:

> In ember-model.js
> <https://github.com/Swrve/ember-model/pull/3#discussion_r40913649>:
>
> >        return this.getHasMany(key, type, meta, this.container);
> > +    },
> > +    set: function(propertyKey, newContentArray, existingArray) {
> > +      //SWRVE-CHANGES
> > +      if (existingArray !== newContentArray) {
>
> this is comparing arrays by reference, is this ok? or do we need to
> compare them based on the content of the array?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Swrve/ember-model/pull/3/files#r40913649>.
>